packages/core/src/service/fileUtil.ts

```diff
@@ -6,6 +6,45 @@ export type RootUriString = `${string}/`
 export type FileExtension = `.${string}`
 
 export namespace fileUtil {
+	/**
+	 * Get the relative URI of `target` based from `base`.
+	 *
+	 * @returns The relative URI, or `undefined` if `target` is not under `base`.
+	 */
+	export function getRelativeUriFromBase(
+		target: string,
+		base: string,
+	): string | undefined {
+		const baseUri = new Uri(base)
+		const targetUri = new Uri(target)
+
+		if (baseUri.origin !== targetUri.origin) {
+			// Different scheme, hostname, and/or port
+			return undefined
+		}
+
+		const baseComponents = baseUri.pathname.split('/')
+			.filter((v) => !!v)
+		const targetComponents = targetUri.pathname.split('/')
+			.filter((v) => !!v)
+
+		for (const [i, c] of baseComponents.entries()) {
+			if (
+				decodeURIComponent(c) !== decodeURIComponent(targetComponents[i])
```

Comment:
I think this will throw if `targetComponents` happens to be shorter than `baseComponents`

Reply:
It doesn't throw but it's weird and won't work if base is `file:///foo/undefined/` and target is `file:///foo/`.

![image](https://github.com/SpyglassMC/Spyglass/assets/15277496/fc539600-fc58-49e4-ad8a-3f114d9122f4)